extraction/src/main/java/io/ssc/trackthetrackers/extraction/resources/GhostDriverExtractor.java

```diff
@@ -0,0 +1,232 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.Scanner;
+
+public class GhostDriverExtractor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GhostDriverExtractor.class);
+
+  private final URLNormalizer urlNormalizer = new URLNormalizer();
+
+
+  public static PhantomJSDriver setup(){
+    Capabilities capabilities = new DesiredCapabilities().phantomjs();
+    // Set PhantomJS Path
+    ((DesiredCapabilities) capabilities).setCapability("phantomjs.binary.path", "/home/felix/Software/phantomjs/bin/phantomjs");
+    ((DesiredCapabilities) capabilities).setCapability("phantomjs.settings.loadImages", false);
+    //((DesiredCapabilities) capabilities).setCapability("phantomjs.settings.localToRemoteUrlAccessEnabled", false);
+
+    return new PhantomJSDriver(capabilities);
+  }
+
+  public synchronized Iterable<Resource> extractResources(String sourceUrl, String html) {
+    return extractResources(sourceUrl, html, null);
+  }
+
+  public synchronized Iterable<Resource> extractResources(String sourceUrl, String html, PhantomJSDriver phantom) {
+
+    Set<Resource> resources = Sets.newHashSet();
+    String prefixForInternalLinks = urlNormalizer.createPrefixForInternalLinks(sourceUrl);
+
+    Document doc = Jsoup.parse(html);
+    Elements iframes = doc.select("iframe[src]");
+    Elements links = doc.select("link[href]");
+    Elements imgs = doc.select("img[src]");
+    Elements scripts = doc.select("script");
+
+    Elements all = iframes.clone();
+    all.addAll(scripts);
+    all.addAll(links);
+    all.addAll(imgs);
+
+    String uri = null;
+
+    List<String> scriptHtml =  new ArrayList<String>();
+    //scriptHtml.add(html);
+
+    for (Element tag : all) {
+      uri = tag.attr("src");
+
+      if (!uri.contains(".")) {
+        uri = tag.attr("href");
+      }
+
+      if (uri.contains(".")) {
+        uri = urlNormalizer.expandIfInternalLink(prefixForInternalLinks, uri);
+        // normalize link
+        try {
+          uri = urlNormalizer.normalize(uri);
+          uri = urlNormalizer.extractDomain(uri);
+        } catch (MalformedURLException e) {
+          if (LOG.isWarnEnabled()) {
+            LOG.warn("Malformed URL: \"" + uri + "\"");
+          }
+        }
+        if (isValidDomain(uri)) {
+          resources.add(new Resource(uri, type(tag.tag().toString())));
+        }
+      }
+
+      /*
+      if (tag.tag().toString().equals("script")) { //filter functions
+        if(tag.data().length() > 1 && tag.data().contains("function")) {
+          scriptHtml.add(tag.data());
+        }
+      }*/
+    }
+
+    scriptHtml.add(html);
+
+
+    for(String shtml : scriptHtml) {
+
+      File temp = null;
+      try {
```

Comment:
sry, my comment was unclear. The scanner that uses the file (and the delete() call on the file) should be done in a finally block